server/services/v1/workers/worker.go

```diff
@@ -0,0 +1,409 @@
+// Copyright 2022-2023 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workers
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/rs/zerolog/log"
+	"github.com/tigrisdata/tigris/schema"
+	"github.com/tigrisdata/tigris/server/metadata"
+	"github.com/tigrisdata/tigris/server/services/v1/database"
+	"github.com/tigrisdata/tigris/server/transaction"
+	ulog "github.com/tigrisdata/tigris/util/log"
+)
+
+const (
+	MAX_ERROR_COUNT = 10
+	LEASE_TIME      = 1 * time.Minute
```

Comment:
`buildIndexTask` in some cases may take an hour when a collection is big, if lease time is small, would it cause another worker to pick up the task? Or somewhere in the code worker is periodically renewing the lease while the building of the index is happening?

Reply:
Right. Each time it does a successful batch index, part of the commit is increasing the lease time for another minute